### PR TITLE
Make migration cancel tests reliable

### PR DESCRIPTION
**Pull request by rmohr**
**What this PR does / why we need it**:

Instead of trying to run stress tests and trying to abort the migration
on an overloaded VMI, create a MigrationPolicy which throttles the
migration speed to 1Mi which creates a huge time window for canceling
the migration.

It works in parallel since we can limit the throtteling to the test
namesapce without changing the global configuration and it even makes the tests
faster because we don't have to wait for the guest to be fully booted.

Fixes tests like

```
Tests Suite: [rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system][sig-compute] VM Live Migration Starting a VirtualMachineInstance live migration cancelation should be able to cancel a migration [sig-storage][storage-req][test_id:2731] with RWX block disk from block volume PVC
```

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/migration_test.go [hunk 1]
```diff
@@ -2248,18 +2249,33 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
...
+					ObjectMeta: metav1.ObjectMeta{
+						Name: util.NamespaceTestDefault,
+						Labels: map[string]string{
+							cleanup.TestLabelForNamespace(util.NamespaceTestDefault): "",
+						},
+					},
+					Spec: v1alpha1.MigrationPolicySpec{
+						BandwidthPerMigration: &quantity,
+						Selectors: &v1alpha1.Selectors{
+							NamespaceSelector: &metav1.LabelSelector{
```
Comment (xpivarc) [inline tests/migration_test.go:2264]: This will affect all vms in the namespace. Wouldn't it be better to use vmi selector? (for parallel run)

**Comment by rmohr** [comment]
/retest

Compute jobs were canceled by prow.

File: tests/libnet/namespace.go [hunk 2]
```diff
@@ -35,7 +36,9 @@ func RemoveAllLabelsFromNamespace(client kubecli.KubevirtClient, namespace strin
 		if ns.Labels == nil {
 			return
 		}
-		ns.Labels = map[string]string{}
+		ns.Labels = map[string]string{
```
[the remark this answers is not in the document]
Comment (rmohr) [reply tests/libnet/namespace.go:39]: The namespaces are just cleaned, not recreated. Therefore we have to keep it here.

File: tests/migration_test.go (hunk 1 above)
Comment (rmohr) [reply tests/migration_test.go:2264]: Every test runs in its own namespace when executed in parallel. NamespaceTestDefault basically has a different name on each ginkgo execution node.

File: tests/libnet/namespace.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (xpivarc) [reply tests/libnet/namespace.go:39]: Ah yes sorry, I forgot we only alter our namespaces and we actually create them for every parallel run.

File: tests/migration_test.go (hunk 1 above)
Comment (xpivarc) [reply tests/migration_test.go:2264]: :+1:  Great that we have migration policies :)

**Review by xpivarc — APPROVED** [review]

File: tests/migration_test.go (hunk 1 above)
Comment (xpivarc) [reply tests/migration_test.go:2264]: Also I should start reading descriptions :)

**Comment by rmohr** [comment]
Migration tests are passing, but running them again to be sure they are fine: 

/test pull-kubevirt-e2e-k8s-1.21-sig-compute-migrations
/test pull-kubevirt-e2e-k8s-1.22-sig-compute-migrations
/test pull-kubevirt-e2e-k8s-1.23-sig-compute-migrations

File: tests/migration_test.go (hunk 1 above)
Comment (xpivarc) [reply tests/migration_test.go:2264]: Do we clean `MigrationPolicy`?

Comment (rmohr) [reply tests/migration_test.go:2264]: Yes, all MigrationPolicies which have the label

```
						Labels: map[string]string{
							cleanup.TestLabelForNamespace(util.NamespaceTestDefault): "",
						},
```

are cleaned when a test in the `NamespaceTestDefault` namespace is done (so only these to not influence each other).

**Review by dhiller — APPROVED** [review]
/approve

**Comment by rmohr** [comment]
/override pull-kubevirt-e2e-k8s-1.21-sig-storage
/override pull-kubevirt-e2e-k8s-1.23-sig-storage

The storage tests fail on a migration test which should only be run on the migration lane. Will be fixed in #7194.

Skipping the checks for now to stablilize the migration tests and bring the test in.

**Pull request merged**